### PR TITLE
Add: add net ip match function

### PR DESCRIPTION
**Pull request by watermelo**
**What this PR does**:
Add match ip and address function

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: net/net_test.go [hunk 1]
```diff
@@ -87,3 +87,81 @@ func TestListenOnUDPRandomPort(t *testing.T) {
 		l.Close()
 	}
 }
+
+func TestMatchIpIpv4Equal(t *testing.T) {
+	flag := MatchIP("192.168.0.1:8080", "192.168.0.1", "8080")
```
Comment (wongoo) [inline net/net_test.go:92]: not need to assign to a variable, assert the result directly

File: net/net.go [hunk 2]
```diff
@@ -175,3 +184,131 @@ func ListenOnUDPRandomPort(ip string) (*net.UDPConn, error) {
 
 	return net.ListenUDP("udp4", &localAddr)
 }
+
+// MatchIP is used to determine whether @pattern and @host:@port match, It's supports subnet/range
+func MatchIP(pattern, host, port string) bool {
```
Comment (wongoo) [inline net/net.go:189]: the param `port` seems redundant for the func，it check whether ip is match

Comment (watermelo) [reply net/net.go:189]: When pattern has port, it will be check.

File: net/net.go [hunk 3]
```diff
@@ -175,3 +184,131 @@ func ListenOnUDPRandomPort(ip string) (*net.UDPConn, error) {
...
+	// if the pattern is subnet format, it will not be allowed to config port param in pattern.
+	if strings.Contains(pattern, "/") {
+		_, subnet, _ := net.ParseCIDR(pattern)
```
[the remark this answers is not in the document]
Comment (watermelo) [reply net/net.go:192]: As a user, pattern means different formats , which will be more convenient for users.

[the remark this answers is not in the document]
Comment (wongoo) [reply net/net.go:192]: does the type of parameter port should be int？

**Pull request merged**